_rules/printable-characters-shortcut-ffbc54.md

```diff
@@ -0,0 +1,591 @@
+---
+id: ffbc54
+name: No keyboard shortcuts use only printable characters
+rule_type: composite
+description: |
+  This rule checks that if keyboard shortcuts are implemented using only printable characters, then there is a mechanism to disable the shortcut, or to remap the shortcut to use one or more non-printable characters keys, or the shortcut for a user interface component is only available when that component has focus.
+accessibility_requirements:
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+input_rules:
+  - 670a30
+  - aa8b52
+  - 1e9941
+acknowledgements:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [HTML document][] with [keyboard shortcuts][keyboard shortcuts].
+
+## Expectation
+
+For the test target, the outcome of at least one of the following rules is passed:
+
+- [Keyboard shortcut can be disabled](https://act-rules.github.io/rules/670a30)
+- [Keyboard shortcut can be remapped](https://act-rules.github.io/rules/aa8b52) 
+- [Keyboard shortcut is only active when component has focus](https://act-rules.github.io/rules/1e9941)
+
+## Assumptions
+
+This rule assumes as applicable [keyboard shortcuts][keyboard shortcuts] those implemented by the test target [content][]. Any other means (e.g. browser extensions, browser settings, user agents, external browser applications) are not considered.
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.1.4: Character Key Shortcuts](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html)
+- [G127 Providing a mechanism to allow users to remap or turn off character key shortcuts](https://www.w3.org/WAI/WCAG21/Techniques/general/G217)
+- [F99 Failure of Success Criterion 2.1.4 due to implementing character key shortcuts that cannot be turned off or remapped](https://www.w3.org/WAI/WCAG21/Techniques/failures/F99)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The [HTML document][] has a [keyboard shortcut][keyboard shortcuts] that uses one [printable][printable characters] and one [non-printable characters][].
```

Comment:
I'd rather say which of the rules is passing, given that the composite rule is only checking if one of the atomic rules is passing.
This gets especially true if the example is already an example of one of the atomic rule. Then it is completely sufficient to just say "rule something is passing" and one can look it to figure out the precise reason.

Reply:
True.
Composite rules currently have descriptions that explain why the corresponding atomic rule is passing. I am not aware of this being a conscious choice or just "the way it is". Example descriptions have been notably bad and are in need of reworking…

---
Why do I think that just saying "this example passes rule FOO" is better?
1. This is in line with the "This [test target] [meets expectations]" format worked in #834. The expectation of the composite rule is to pass some atomic rule, not to have a shortcut with non-printable character.
2. This emphasises the fact that composite rules know nothing about input aspects and can only care about the outcomes of atomic rules. For the reader, this reminds that point. For the writer, if you cannot write the example description like that, it is a strong indication that the composite rule is poorly written.
3. This is better future-proof for composite rules with more complex logic. A composite rule could be "each test target either passes rule A; or passes both rules B and C" or "each test target that passes rule A also passes rule B", … Writing example descriptions that go into the details of each atomic rule would get super-hairy with complex composite rules. Writing descriptions that just refer to rules outcome and the logic would stay fairly easy.

---
Of course, your mileage may vary. We have not really discussed how to uniformise example descriptions in composite rules. That's why my initial comment is written as a personal opinion and not as a strong request (and I won't block the PR if that's the only thing left).